From d3e16f248ea60925ae4498a670797dbb5154939f Mon Sep 17 00:00:00 2001
From: Christopher Champagne <cjchampagne@habana.ai>
Date: Fri, 23 May 2025 15:59:11 +0300
Subject: [PATCH] [SW-179783] g3: sensord: add read rate defines

Add defines for the read rates of sensord sensor entries.

Change-Id: Iee88adfc6b7fa25ea00110a68e70c968cfbba5f8
Signed-off-by: Christopher Champagne <cjchampagne@habana.ai>
---

diff --git a/include/sensord.h b/include/sensord.h
index ff2db38..417bf99 100644
--- a/include/sensord.h
+++ b/include/sensord.h
@@ -10,6 +10,11 @@
 
 #include <pldm_base.h>
 
+/* Rate value in msec */
+#define SENSOR_SAMPLE_RATE_LOW		CONFIG_SENSORD_POLLING_INTERVAL_MSEC
+#define SENSOR_SAMPLE_RATE_HIGH		2500
+#define SENSOR_SAMPLE_RATE_VERY_HIGH	1000
+
 struct sensord_sensor_value_single {
 	uint32_t present_reading;
 	struct pldm_pmc_threshold_data threshold_values;
diff --git a/src/sensord.c b/src/sensord.c
index fc76704..df88a3e 100644
--- a/src/sensord.c
+++ b/src/sensord.c
@@ -388,7 +388,7 @@
 				    struct sensord_sensor_entry *sensor,
 				    int64_t current_time)
 {
-	k_timeout_t default_polling_interval = K_MSEC(CONFIG_SENSORD_POLLING_INTERVAL_MSEC);
+	k_timeout_t default_polling_interval = K_MSEC(SENSOR_SAMPLE_RATE_LOW);
 
 	state->timestamp_next_poll = current_time +
 				     (sensor->api.polling_rate_ms ?
